arch/arm/src/s32k3xx/hardware/s32k3xx_qspi.h

```diff
@@ -0,0 +1,504 @@
+#define QSPI_BUFCR_MSTRID_MASK            (0x0f << QSPI_BUFCR_MSTRID_SHIFT)
+#define QSPI_BUFCR_MSTRID(n)              ((n << QSPI_BUFCR_MSTRID_SHIFT) & QSPI_BUFCR_MSTRID_MASK)
+                                                    /* Bits 4-7: Reserved */
+#define QSPI_BUFCR_ADATSZ_SHIFT           (8)       /* Bits 8-13: AHB data transfer size (ADATSZ) */
+#define QSPI_BUFCR_ADATSZ_MASK            (0x3f << QSPI_BUFCR_ADATSZ_SHIFT)
+#define QSPI_BUFCR_ADATSZ(n)              ((n << QSPI_BUFCR_ADATSZ_SHIFT) & QSPI_BUFCR_ADATSZ_MASK)
+                                                    /* Bits 14-31: Reserved */
+#define QSPI_BUF3CR_ALLMST                (1 << 31) /* Bit 31: All master enable (ALLMST) */
+
+/* Buffer Generic Configuration Register (BFGENCR) */
+
+                                                    /* Bits 0-11: Reserved */
+#define QSPI_BFGENCR_SEQID_SHIFT          (12)      /* Bits 12-15: Points to a sequence in the LUT (SEQID) */
+#define QSPI_BFGENCR_SEQID_MASK           (0x0f << QSPI_BFGENCR_SEQID_SHIFT)
+                                                    /* Bits 16-31: Reserved */
+
+/* SOC Configuration Register (SOCCR) */
+
+#define QSPI_SOCCR_SOCCFG_SHIFT           (0)       /* Bits 0-31: SOC configuration (SOCCFG) */
+#define QSPI_SOCCR_SOCCFG_MASK            (0xffffffff << QSPI_SOCCR_SOCCFG_SHIFT)
+#define QSPI_SOCCR_OBE_PULL_TMG_RLX       (1 << 0) /* Bit 0: obe_pull_timing_relax_b*/
+#define QSPI_SOCCR_IBE                    (1 << 1) /* Bit 1: IBE */
+#define QSPI_SOCCR_OBE                    (1 << 2) /* Bit 2: OBE */
+#define QSPI_SOCCR_DSE                    (1 << 3) /* Bit 3: DSE */
+#define QSPI_SOCCR_PUE                    (1 << 4) /* Bit 4: PUE */
+#define QSPI_SOCCR_PUS                    (1 << 5) /* Bit 5: PUS */
+#define QSPI_SOCCR_SRE                    (1 << 6) /* Bit 6: SRE */
+
+/* Buffer n Top Index Register (BUFnIND) */
+
+                                                    /* Bits 0-2: Reserved */
+#define QSPI_BUFIND_TPINDX_SHIFT          (3)       /* Bits 3-8: Top index of buffer n (TPINDXn) */
+#define QSPI_BUFIND_TPINDX_MASK           (0x3f << QSPI_BUFIND_TPINDX_SHIFT)
+                                                    /* Bits 9-31: Reserved */
+
+/* DLL Flash Memory A Configuration Register (DLLCRA) */
+
+#define QSPI_DLLCRA_SLV_UPD               (1 << 0)  /* Bit 0: Slave update (SLV_UPD) */
+#define QSPI_DLLCRA_SLV_DLL_BYPASS        (1 << 1)  /* Bit 1: Slave DLL bypass (SLV_DLL_BYPASS) */
+#define QSPI_DLLCRA_SLV_EN                (1 << 2)  /* Bit 2: Slave enable (SLV_EN) */
+                                                    /* Bits 3-7: Reserved */
+
+#define QSPI_DLLCRA_SLV_DLY_COARSE_SHIFT  (8)       /* Bits 8-11: Delay elements in each delay tap (SLV_DLY_COARSE) */
+#define QSPI_DLLCRA_SLV_DLY_COARSE_MASK   (0x0f << QSPI_DLLCRA_SLV_DLY_COARSE_SHIFT)
+#define QSPI_DLLCRA_SLV_DLY_COARSE(n)     ((n << QSPI_DLLCRA_SLV_DLY_COARSE_SHIFT) & QSPI_DLLCRA_SLV_DLY_COARSE_MASK)
```

Comment:
```suggestion
#define QSPI_DLLCRA_SLV_DLY_COARSE(n)     (((n) << QSPI_DLLCRA_SLV_DLY_COARSE_SHIFT) & QSPI_DLLCRA_SLV_DLY_COARSE_MASK)
```
Could you please change here and similar places?

Reply:
I've got them all now I think, thanks for being sharp.